packages/prover/src/utils/consensus.ts

```diff
@@ -6,14 +6,21 @@ import {Logger} from "@lodestar/utils";
 import {MAX_PAYLOAD_HISTORY} from "../constants.js";
 import {hexToBuffer} from "./conversion.js";
 
+export async function fetchBlock(api: Api, slot: number): Promise<capella.SignedBeaconBlock | undefined> {
+  const res = await api.beacon.getBlockV2(slot);
```

Comment:
are we fine with throwing here if there is a network error?

Reply:
If an error is thrown here then a light-client header is not processed (that error will be logged).

I think that it's ok in the context of this PR as it doesn't change this behavior. If required a proper solution could be implemented separately.